### PR TITLE
Improved project root Makefile

### PR DESCRIPTION
**Pull request by johnbaldwin**
Reworked Figures Makefile to make it better to use. Still an evolving effort, just wanted to get it into master

@ahmedaljazzar I added you because I took your ideas from the makefile you made for the cloud devstack

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -26,6 +26,7 @@ clean_python:
 	find devsite -type f -name "*.pyc" -exec rm -f {} \;
 	find tests -type d -name __pycache__ -exec rm -r {} \+
 	find devsite -type d -name __pycache__ -exec rm -r {} \+
+<<<<<<< Updated upstream
```
Comment (OmarIthawi) [inline Makefile]:
```suggestion
```

File: Makefile [hunk 2]
```diff
@@ -45,6 +46,49 @@ coverage:
 
 lint:
 	pylint --load-plugins pylint_django ./figures
+=======
```
Comment (OmarIthawi) [inline Makefile]:
```suggestion
```

File: Makefile [hunk 3]
```diff
@@ -45,6 +46,49 @@ coverage:
...
+	@make ve/bin/figures-ws
+
+ve.coverage: ve/bin/figures-ws  ## coverage with the built-invirtualenv
+	@. ve/bin/activate; coverage run --source figures -m py.test; coverage report -m
+
+ve.test: python.clean ve/bin/figures-ws ## Run pytest using the built-in virtualenv
+	# We need to have the packages installed in the ./requirements.txt
+	# For tests to run. Best to do this in a venv
+	@. ve/bin/activate; py.test
+>>>>>>> Stashed changes
```
Comment (OmarIthawi) [inline Makefile]:
```suggestion
```

File: Makefile [hunk 4]
```diff
@@ -80,3 +124,29 @@ update_mock_migrations:
 	./manage.py makemigrations course_overviews && \
 	./manage.py makemigrations student
 
+<<<<<<< Updated upstream
```
Comment (OmarIthawi) [inline Makefile]:
```suggestion
```

File: Makefile [hunk 5]
```diff
@@ -80,3 +124,29 @@ update_mock_migrations:
...
+=======
```
Comment (OmarIthawi) [inline Makefile]:
```suggestion
```

File: Makefile [hunk 6]
```diff
@@ -80,3 +124,29 @@ update_mock_migrations:
...
+## Frontend targets
+##
+
+frontend.build.clean:  ## Clean the webpack build
+	rm -rf figures/static/figures/*
+	rm figures/webpack-stats.json
+
+frontend.build: frontend.build.clean  ## Yarn build frontend assets
+	cd frontend; yarn build
+>>>>>>> Stashed changes
```
Comment (OmarIthawi) [inline Makefile]:
```suggestion
```

**Comment by johnbaldwin** [comment]
@OmarIthawi Thanks! I had saved a copy and pushed the wrong one here. Will update (which will re-request PR)

**Comment by johnbaldwin** [comment]
@OmarIthawi I pushed the right one this time

File: Makefile [hunk 7]
```diff
@@ -1,74 +1,88 @@
...
 
-pip_install:
-	pip install -r devsite/requirements/hawthorn.txt
+coverage:  ## Run coverage, without the built-in virtualenv
+	@. ve/bin/activate; coverage run --source figures -m py.test; coverage report -m
+
+
+### Automatically constructed Virtualenv based targets
+
+ve/bin/figures-ws: devsite/requirements/${EDX_PLATFORM_RELEASE}_appsembler.txt
```
Comment (OmarIthawi) [inline Makefile]: Make magic! I didn't know it can work on paths. Thanks!

Comment (johnbaldwin) [reply Makefile]: Cool, huh @OmarIthawi? I learned that one from @ahmedaljazzar

**Pull request merged**